core/src/test/java/org/testcontainers/dockerclient/EnvironmentAndSystemPropertyClientProviderStrategyTest.java

```diff
@@ -0,0 +1,100 @@
+package org.testcontainers.dockerclient;
+
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.LocalDirectorySSLConfig;
+import com.github.dockerjava.transport.SSLConfig;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.testcontainers.utility.MockTestcontainersConfigurationRule;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isNull;
+
+/**
+ * Test that we can use Testcontainers configuration file to override settings. We assume that docker-java has test
+ * coverage for detection of environment variables (e.g. DOCKER_HOST) and its own properties config file.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class EnvironmentAndSystemPropertyClientProviderStrategyTest {
+
+    @Rule
+    public MockTestcontainersConfigurationRule mockConfig = new MockTestcontainersConfigurationRule();
+    private URI defaultDockerHost;
+    private com.github.dockerjava.core.SSLConfig defaultSSLConfig;
+
+    @Before
+    public void checkEnvironmentClear() {
+        // If docker-java picks up non-default settings from the environment, our test needs to know to expect those
+        DefaultDockerClientConfig defaultConfig = DefaultDockerClientConfig.createDefaultConfigBuilder().build();
+        defaultDockerHost = defaultConfig.getDockerHost();
+        defaultSSLConfig = defaultConfig.getSSLConfig();
+    }
+
+    @Test
+    public void testWhenConfigAbsent() {
+        Mockito.doReturn(null).when(TestcontainersConfiguration.getInstance()).getEnvVarOrUserProperty(eq("docker.host"), isNull());
+        Mockito.doReturn(null).when(TestcontainersConfiguration.getInstance()).getEnvVarOrUserProperty(eq("docker.tls.verify"), isNull());
+        Mockito.doReturn(null).when(TestcontainersConfiguration.getInstance()).getEnvVarOrUserProperty(eq("docker.cert.path"), isNull());
+
+        EnvironmentAndSystemPropertyClientProviderStrategy strategy = new EnvironmentAndSystemPropertyClientProviderStrategy();
+
+        TransportConfig transportConfig = strategy.getTransportConfig();
+        assertEquals(defaultDockerHost, transportConfig.getDockerHost());
+        assertEquals(defaultSSLConfig, transportConfig.getSslConfig());
+
+        String dockerHostIpAddress = strategy.getDockerHostIpAddress();
+        assertEquals("localhost", dockerHostIpAddress);
```

Comment:
depending on `defaultDockerHost`, it may not be `localhost`

Also, I think we don't need to assert it here, as we're only checking what's returned as `getTransportConfig()`, while the host IP detection is a separate concern

Reply:
I wanted to ensure we're not borking the setting, e.g. setting it to `"null"`. Looks like I failed to make it look at the default in 7b42ea7a7d4db739847b68ac2fd0feab32a7b5f2, so will add.

Reply:
Have removed x3